rust/extns/numaflow-sqs/src/sink.rs

```diff
@@ -0,0 +1,638 @@
+}
+
+/// Builder for creating and configuring an SQS sink.
+#[derive(Clone)]
+pub struct SqsSinkBuilder {
+    config: SqsSinkConfig,
+    client: Option<Client>,
+}
+/// Response from sending a message to SQS.
+#[derive(Debug)]
+pub struct SqsSinkResponse {
+    /// ID of the message that was sent
+    pub id: String,
+    /// Status of the send operation
+    pub status: Result<()>,
+    /// Error code if any
+    pub code: Option<String>,
+    /// Indicates if the error was caused by the sender
+    pub sender_fault: Option<bool>,
+}
+
+impl Default for SqsSinkBuilder {
+    fn default() -> Self {
+        Self::new(SqsSinkConfig {
+            region: SQS_DEFAULT_REGION.to_string(),
+            queue_name: "".to_string(),
+            queue_owner_aws_account_id: "".to_string(),
+        })
+    }
+}
+
+impl SqsSinkBuilder {
+    pub fn new(config: SqsSinkConfig) -> Self {
+        Self {
+            config,
+            client: None,
+        }
+    }
+
+    pub fn config(mut self, config: SqsSinkConfig) -> Self {
+        self.config = config;
+        self
+    }
+
+    pub fn client(mut self, client: Client) -> Self {
+        self.client = Some(client);
+        self
+    }
+
+    pub async fn build(self) -> Result<SqsSink> {
+        let sqs_client = match self.client {
+            Some(client) => client,
+            None => create_sqs_client(Some(self.config.clone())).await?,
+        };
+
+        let queue_name = self.config.queue_name.clone();
+        let queue_owner_aws_account_id = self.config.queue_owner_aws_account_id.clone();
+
+        let get_queue_url_output = sqs_client
+            .get_queue_url()
+            .queue_name(queue_name)
+            .queue_owner_aws_account_id(queue_owner_aws_account_id)
+            .send()
+            .await
+            .map_err(|err| Error::Sqs(err.into()))?;
+
+        let queue_url = get_queue_url_output
+            .queue_url
+            .ok_or_else(|| Error::Other("Queue URL not found".to_string()))?;
+
+        tracing::info!(queue_url = queue_url.clone(), "Queue URL found");
+
+        let (handler_tx, handler_rx) = mpsc::channel(10);
+        tokio::spawn(async move {
+            let mut actor = SqsSinkActor::new(handler_rx, sqs_client, queue_url);
+            actor.run().await;
+        });
+
+        Ok(SqsSink {
+            actor_tx: handler_tx,
+        })
+    }
+}
+
+impl SqsSink {
+    /// Sends a batch of messages to the SQS queue.
+    ///
+    /// Returns responses for each message, including success or failure status.
+    pub async fn sink_messages(
+        &self,
+        messages: Vec<SqsSinkMessage>,
+    ) -> Result<Vec<SqsSinkResponse>> {
+        let (tx, rx) = oneshot::channel();
+        let msg = SqsSinkActorMessage::SendMessageBatch {
+            respond_to: tx,
+            messages,
+        };
+        if (self.actor_tx.send(msg).await).is_err() {
+            tracing::error!("Failed to send message to SqsSinkActor");
+        }
```

Comment:
Do we really need an actor? We're only performing one operation at a time, and it's a blocking call. Also, the SQS client is cloneable, you can invoke `write_to_batch` directly from here.